### DISCUSSION TITLE
Error updating Google Cloud SQL (MySQL)

### DISCUSSION THREAD

**Discussion by jgotting** [Budibase/budibase#3204] [Help]
Hi,
great job with Budibase.

I've successfully connected a Google CloudSQL database with the MySQL-connector. App is in Budibase Cloud.
Creating a new record from Budibase's data view works well. However, when I try to update an existing record in the Data view, I get the following error message. I should also mention that updating works well from MySQL Workbench for example.

"%5B32%5D does not exist"

![image](https://user-images.githubusercontent.com/10381252/139582478-457e6143-7834-48ff-9759-8fe666970cc6.png)

 BR/Johan

**Comment by mjashanks** [comment]
Hey Johan.

I have not been able to reproduce this so far. 

- Does your table have a primary key set? 
- What version of Budibase are you using? You can find this in the `Updates` section on self-hosted - or just say `Cloud` if that's what it is!
- Do you have any relationships set you your table?

It would be useful to see the create scripts for your table & any related tables.

Thanks!

**Reply by jgotting** [reply]
Hi Michael,

thanks for your reply!

Primary key: YES, 
Budibase Cloud, latest version (0.9.173)
No relationships, only one table (apart from the internal user table)

/* CREATE TABLE */
CREATE TABLE impressions(
id_impression INT NOT NULL AUTO_INCREMENT,
url VARCHAR(100),
description VARCHAR(100),
datetime1 VARCHAR(100),
preview VARCHAR(200),
original VARCHAR(200),
change1 VARCHAR(100),
view_changes VARCHAR(200),
company VARCHAR(100),
category VARCHAR(100),
content_type VARCHAR(50) NULL,
validity TINYINT NULL DEFAULT 1,
PRIMARY KEY (`id_impression`)
);

Insert statement (if needed) in private DM.

Thanks//Johan

**Reply by jgotting** [reply]
Hi Michael,
just wanted to reach out and check if you received my answers to your questions.
BR/Johan

**Reply by mjashanks** [reply]
Yes, I did thanks. I have setup your table on a local MySQL instance and it's working ok for me. My next step was to try it on Google Cloud, which I haven't got around to yet!

**Reply by jgotting** [reply]
Hi again,
thank you. In order to reproduce on Google Cloud I provide some configuration data.

- Database version is MySQL 5.7
- Auto storage increase is enabled
- Automated backups are enabled
- Stored in Multi-region: us - Data centers in the United States
- Point-in-time recovery is enabled
- Located in europe-north1-a
- Highly available (regional)
- No database flags set
- No labels set

Br//Johan

**Reply by mjashanks** [reply]
Just got around to reproducing this on Google Cloud. I have logged an issue for it here: https://github.com/Budibase/budibase/issues/3258 - we will look at it as soon as we can.

It likely only happens with MySQL 5.7. As above, I cannot reproduce on v8 (but running locally).

**Reply by jgotting** [reply]
Great, thanks!
